File: core/universal/structured_payload_v3_engine.py
from __future__ import annotations
import re

def parse_structured_payload(text: str):
    src = text or ''
    return {
        "has_openapi": bool(re.search(r'openapi\s*:\s*3', src, flags=re.I)),
        "has_graphql": 'type Query' in src or 'schema {' in src,
        "has_dockerfile": 'FROM ' in src,
        "has_ci": any(k in src for k in ['github/workflows', 'gitlab-ci', 'Jenkinsfile']),
    }

File: src/universal/structuredPayloadV3Engine.ts
/**
 * Converted from Python: core/universal/structured_payload_v3_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function parseStructuredPayload(text: any): any {
  var src: any = py.or2(text, () => (""));
  return {"has_openapi": py.truthy(py.reSearch("openapi\\s*:\\s*3", src, "i")), "has_graphql": py.or2(py.contains(src, "type Query"), () => (py.contains(src, "schema {"))), "has_dockerfile": py.contains(src, "FROM "), "has_ci": py.any(py.iter(["github/workflows", "gitlab-ci", "Jenkinsfile"]).map((k: any) => py.contains(src, k)))};
}
